coreneuron/utils/memory_utils.cpp

```diff
@@ -70,12 +73,26 @@ double nrn_mallinfo(void) {
         KERN_SUCCESS)
         return (size_t)0L; /* Can't access? */
     return info.resident_size / (1024.0 * 1024.0);
-#elif defined HAVE_MALLOC_H
-    struct mallinfo m;
-    m = mallinfo();
-    mbs = (m.hblkhd + m.uordblks) / (1024.0 * 1024.0);
+#else
+    std::ifstream file;
+    std::stringstream buffer;
+    file.open("/proc/self/statm");
+    if (file.is_open()) {
+        buffer << file.rdbuf();
+        file.close();
+        unsigned long long int total_virtual_memory;
+        buffer >> total_virtual_memory;
+        mbs = (total_virtual_memory * sysconf(_SC_PAGESIZE)) / (1024.0 * 1024.0);
```

Comment:
I am wondering if we should use VM or RSS. (e.g. in hpctools  we considered only RSS. While using mallinfo I have seen that considering `mmap` and shared library related memory usage gives very different numbers).

How your measurements look for RSS vs VM?